server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -360,6 +360,12 @@ private void cancel(String externalCompactionId) throws TException {
     }
   }
 
+  @Override
+  public void cancel(TInfo tinfo, TCredentials credentials, String externalCompactionId)
+      throws TException {
```

Comment:
The other thrift api methods perform a security check first, do we want to do that here?

Reply:
I didn't think it needed SYSTEM level permission (root) like the other method required. I thought it should require at least some level of USER permission but wasn't sure since the permissions are so confusing. Any ideas?

Reply:
oh ok. I'm not up to speed on all of the permissions, I just know I used it on all of the other thrift api methods. If it's not appropriate, then that is good with me.

Reply:
I saw that the FATE Table cancel compact operation has the same permissions as a user needs to compact. So we could try that:
From FateServiceHandler:
`          canCancelCompact = manager.security.canCompact(c, tableId, namespaceId);
`